code/game/gamemodes/changeling/cellular_emporium.dm

```diff
@@ -0,0 +1,81 @@
+// cellular emporium
+// The place where changelings go to buy their biological weaponry.
+
+/datum/cellular_emporium
+	var/name = "cellular emporium"
+	var/datum/changeling/changeling
+
+/datum/cellular_emporium/New(my_changeling)
+	changeling = my_changeling
+
+/datum/cellular_emporium/Destroy()
+	changeling = null
+	. = ..()
+
+/datum/cellular_emporium/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/ui_state/state = always_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "cellular_emporium", name, 900, 480, master_ui, state)
+		ui.open()
+
+/datum/cellular_emporium/ui_data(mob/user)
+	var/list/data = list()
+
+	var/can_readapt = changeling.canrespec
+	var/genetic_points_remaining = changeling.geneticpoints
+	var/absorbed_dna_count = changeling.absorbedcount
+
+	data["can_readapt"] = can_readapt
+	data["genetic_points_remaining"] = genetic_points_remaining
+	data["absorbed_dna_count"] = absorbed_dna_count
+
+	var/list/abilities = list()
+
+	for(var/path in subtypesof(/obj/effect/proc_holder/changeling))
+		var/obj/effect/proc_holder/changeling/ability = new path()
```

Comment:
You are correct, I'll get it initial()'d. (I was just copying what the old code did, honest)